stl/inc/source_location

```diff
@@ -8,8 +8,16 @@
 #define _SOURCE_LOCATION_
 #include <yvals_core.h>
 #if _STL_COMPILER_PREPROCESSOR
-#ifndef __cpp_consteval
+#if !defined(__cpp_lib_source_location)
+#if defined(__clang__)
+#if __clang_major__ < 15
+#pragma message("Your clang is too old. We need at least clang 15.0.0")
+#else
+#pragma message("The contents of <source_location> are available only with C++20.")
+#endif
+#else
 #pragma message("The contents of <source_location> are available only with C++20 consteval support.")
+#endif
 #else // ^^^ !defined(__cpp_consteval) / defined(__cpp_consteval) vvv

```

Comment:
Since the conditions of the preprocessor `#if`s changed, update them accordingly.
```suggestion
#else // ^^^ !defined(__cpp_lib_source_location) / defined(__cpp_lib_source_location) vvv
```